src/components/link/index.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { ElementType, FC, ReactNode } from 'react';
+import { chakra, useMultiStyleConfig } from '@chakra-ui/react';
+
+interface Props {
+  children: ReactNode;
+  as?: ElementType;
+  leftIcon?: ElementType;
+  rightIcon?: ElementType;
+  isDisabled?: boolean;
+  isExternal?: boolean;
+  [key: string]: unknown;
```

Comment:
To fit all router libraries, the dynamic prop is necessary. In future, we can scope our library usage, we can then replace it by specified attributes